### PR TITLE
fix: better debug message for missing network times

### PR DESCRIPTION
**Pull request by patrickhulce**
Improves the debug information when network throughput can't be computed and defaults the result to Infinity instead of 0 which makes more sense since the only use of networkThroughput is to be the divisor (the computed artifact is also used by total byte weight which makes sense to say 0 ms time and still report the byte savings)

This was affecting chrome URLs which don't have any timing information, results from before and after for `chrome://version` below (the optimized images error is for tainting canvas which apparently all chrome images do)

**Before**
image

**After**
image

### CODE REVIEW

File: lighthouse-core/gather/computed/network-throughput.js [hunk 1]
```diff
@@ -38,6 +36,10 @@ class NetworkThroughput extends ComputedArtifact {
       return boundaries;
     }, []).sort((a, b) => a.time - b.time);
 
+    if (!timeBoundaries.length) {
```
Comment (brendankenny) [inline lighthouse-core/gather/computed/network-throughput.js]: can you document this in the jsdoc?

File: lighthouse-core/report/v2/renderer/category-renderer.js [hunk 2]
```diff
@@ -193,6 +193,12 @@ class CategoryRenderer {
     const descriptionEl = this._dom.createChildOf(element, 'div', 'lh-perf-hint__description');
     descriptionEl.appendChild(this._dom.convertMarkdownLinkSnippets(audit.result.helpText));
 
+    if (audit.result.debugString) {
```
Comment (brendankenny) [inline lighthouse-core/report/v2/renderer/category-renderer.js]: what is this adding that wasn't there before? Not clear from the screenshot

File: lighthouse-core/test/audits/byte-efficiency/byte-efficiency-audit-test.js [hunk 3]
```diff
@@ -85,7 +85,7 @@ describe('Byte efficiency base audit', () => {
         {wastedBytes: 450, totalBytes: 1000, wastedPercent: 50},
         {wastedBytes: 400, totalBytes: 450, wastedPercent: 50},
       ],
-    });
+    }, 1000);
```
Comment (brendankenny) [inline lighthouse-core/test/audits/byte-efficiency/byte-efficiency-audit-test.js]: need new tests, too

File: lighthouse-core/gather/computed/network-throughput.js (hunk 1 above)
Comment (patrickhulce) [inline lighthouse-core/gather/computed/network-throughput.js]: done

File: lighthouse-core/test/audits/byte-efficiency/byte-efficiency-audit-test.js (hunk 3 above)
Comment (patrickhulce) [inline lighthouse-core/test/audits/byte-efficiency/byte-efficiency-audit-test.js]: done

File: lighthouse-core/report/v2/renderer/category-renderer.js (hunk 2 above)
Comment (patrickhulce) [inline lighthouse-core/report/v2/renderer/category-renderer.js]: it was, but not if it didn't completely fail (this was sort of drive by, there are other debugStrings that can be set)

Comment (patrickhulce) [inline lighthouse-core/report/v2/renderer/category-renderer.js]: added a test now for it

Comment (brendankenny) [inline lighthouse-core/report/v2/renderer/category-renderer.js]: > it was, but not if it didn't completely fail

it was...what? :P

**Comment by patrickhulce** [comment]
> a new test in network-throughput-test.js to replicate what was coming in from chrome:// URLs would be good too, but otherwise LGTM

done

**Comment by brendankenny** [comment]
travis is greeeeeen

**Pull request merged**